### PR TITLE
chore: PassageError now correctly parses ApiException's deserialized data and raw body

### PR DESCRIPTION
**Pull request by ctran88**
## What's New?

- Fixes issue where `PassageError` was parsing the wrong error field from `ApiResponse`
- Parses `data` if the error came from our OpenAPI schema
- Parses `body` if the error came was not from our OpenAPI schema
- title is `chore` so this doesn't show up in the release notes since this is a bug fix for a feat that will already be in the notes

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have manually tested my code thoroughly
- [ ] I have added/updated inline documentation for public facing interfaces if relevant
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing integration and unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: passageidentity/errors.py [hunk 1]
```diff
@@ -39,8 +39,13 @@ def __init__(
     @classmethod
     def from_response_error(cls, response_error: ApiException, message: str | None = None) -> PassageError:
         """Initialize the error with a response body and optional message."""
-        error_code = response_error.body["code"] if response_error.body else None
-        error_msg = response_error.body["error"] if response_error.body else None
```
Comment (vanessa-passage) [inline passageidentity/errors.py:43]: Breaking change if `error_msg` is no longer returned.

File: tests/errors_test.py [hunk 2]
```diff
@@ -29,26 +33,41 @@ def test_error_with_only_message() -> None:
...
+        data=None,
+        body="some error",
+    )
+
+    error = PassageError.from_response_error(response_error)
+    assert error.message == "some error"
+    assert error.status_code == 400
+    assert error.error_code is None
     assert error.status_text is None
     assert error.error is None
```
Comment (vanessa-passage) [inline tests/errors_test.py:73]: Are we testing for no response only passing in the message? Ex: https://github.com/passageidentity/passage-python/blob/32acfea6a26407ddba235741182c4bd2548e8aaf/passageidentity/helper.py#L23

File: passageidentity/errors.py (hunk 1 above)
Comment (ctran88) [reply passageidentity/errors.py:43]: `error_msg` was never returned. it's only used as a local var to concat into `msg`, which is set and returned in all cases.

also, this method is completely new and only used by the new methods, so it's not touching any existing behavior in the latest published release.

File: tests/errors_test.py (hunk 2 above)
Comment (ctran88) [reply tests/errors_test.py:73]: https://github.com/passageidentity/passage-python/blob/6016c2d574117755bce753d6de1680e35015c45c/tests/errors_test.py#L24

**Review by tdeshong — APPROVED** [review]

**Pull request merged**